### PR TITLE
fix(clerk-js,types): Fix typescript errors

### PR DESCRIPTION
**Pull request by desiprisg**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

### CODE REVIEW

File: packages/clerk-js/src/core/clerk.ts [hunk 1]
```diff
@@ -502,7 +503,7 @@ export default class Clerk implements ClerkInterface {
       session = (this.client.sessions.find(x => x.id === session) as ActiveSessionResource) || null;
     }
 
-    let newSession = session === undefined ? this.session : session;
+    let newSession = session ? session : this.session || null;
```
Comment (SokratisVidros) [inline packages/clerk-js/src/core/clerk.ts:506]: This change might look innocent but I am sure it's not. In ClerkJS we have three states for sessions

A. Session is set -> Signed in
B. Null -> Signed out
C. Undefined -> Transitioning from signed out to signed in 

That is this change might have negative side effects.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

File: packages/clerk-js/src/core/clerk.ts (hunk 1 above)
Comment (desiprisg) [reply packages/clerk-js/src/core/clerk.ts:506]: I was influenced by the commend under this. We used to have an `as ActiveSessionResource | null` there but we removed it by accident (we didn't see it caused a typescript error.

Comment (desiprisg) [reply packages/clerk-js/src/core/clerk.ts:506]: https://github.com/clerkinc/javascript/pull/1042/files/ce824451e6cb28619825e3fcefbc154e90695ce5#diff-1952d6b8ac4b5495e1dba57c32bff9ce72d43f83595c1bc8532a76cf31feb753R508

Comment (desiprisg) [reply packages/clerk-js/src/core/clerk.ts:506]: This is causing the error to fail anyway, so we'll take a closer look!

Comment (dimkl) [reply packages/clerk-js/src/core/clerk.ts:506]: I wouldn't change this line of code. 😅

Comment (nikosdouvlis) [reply packages/clerk-js/src/core/clerk.ts:506]: As everyone said, these 2 lines are not equivalent unfortunately! 
We only want to fallback to `this.session` when `session` is undefined, as `undefined` and `null` have a significant meaning for this flow. This is not your fault of course, we need to revisit the `undefined / null / session` approach

**Review by nikosdouvlis — COMMENTED** [review]
Let's also remove the `( ... || true)` from the `npm run build` script before this gets merged :)

File: packages/clerk-js/src/core/clerk.ts (hunk 1 above)
Comment (panteliselef) [reply packages/clerk-js/src/core/clerk.ts:506]: Should we add a comment highlighting A,B,C as Sokratis mentioned ?

File: packages/clerk-js/src/core/clerk.ts [hunk 2]
```diff
@@ -685,49 +686,56 @@ export default class Clerk implements ClerkInterface {
       { base: getClerkQueryParam(CLERK_SATELLITE_URL) as string, searchParams },
       { stringify: true },
     );
-    return this.navigate(this.buildUrlWithAuth(backToSatelliteUrl, false));
+    return this.navigate(this.buildUrlWithAuth(backToSatelliteUrl));
```
Comment (panteliselef) [inline packages/clerk-js/src/core/clerk.ts:689]: 💯

**Review by nikosdouvlis — APPROVED** [review]
Excellent work @desiprisg

**Pull request merged**